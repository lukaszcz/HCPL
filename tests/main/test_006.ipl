
// test optionality of separators

let x = 5
let y = 7

let a = x + y;

print a;

let z = x * y

syntax operator aaa is left, binary, after or
syntax operator bbb is left, binary, prio aaa
syntax operator ccc is left, binary, prio bbb

let aaa = \x \y . x + y
let bbb = \x \y . x * y
let ccc = \x \y . x - y
in
print (3 aaa 4);
print (z bbb 2);
print (z ccc 2);
let ccc = \x \y . x div y
in
print (z ccc 5);
