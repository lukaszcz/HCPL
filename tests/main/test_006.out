12
7
70
33
7
